srcpkgs/xbps-triggers/files/dkms

```diff
@@ -30,14 +30,24 @@ remove_modules() {
 			if $(echo "$line" | egrep -vq '(added|built|installed)'); then
 				shift 2; continue
 			fi
-			_modver=$(echo "$line"|sed "s/$1,[[:blank:]]\([^,]*\)[,:].*/\1/;t;d")
-			_kver=$(echo "$line"|awk '{print $3}'|sed "s/\(.*\),$/\1/")
-			echo -n "Removing DKMS module '${1}-${_modver}' for kernel-${_kver}... "
-			$DKMS remove -m "$1" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
-			if [ $? -eq 0 ]; then
-				echo "done."
+			_kver=$(echo "$line"|awk '{print $2}'); _kver=${_kver%,}
+			_modver=$(echo "$line"|awk '{print $1}'|sed 's#^.*/\(.*\).$#\1#;')
+			if [ "$_kver" = added ] && [ -d "/var/lib/dkms/${1}/${_modver}/build" ]; then
+				echo -n "Cleaning up unbuilt DKMS module '${1}-${_modver}'... "
+				rm -rf "/var/lib/dkms/${1}/${_modver}/build"
+				if [ $? -eq 0 ]; then
+					echo "done."
+				else
+					echo "FAILED!"
+				fi
```

Comment:
For a demo of this code
- Ensure you are on linux `5.15` or greater
- Install `ddcci-dkms-0.4.1` (from any commit before #37127 is merged)
  (During installation, you should see this message)
  ```
  ddcci-dkms-0.4.1_1: configuring ...
  DKMS module 'ddcci-0.4.1' already added, skipping.
  Building DKMS module 'ddcci-0.4.1' for kernel-5.xx.yy_z... FAILED!
  DKMS module 'ddcci-0.4.1' failed to build, please check /var/lib/dkms
  for errors in the log file.
  ```
- Run `xbps-remove ddcci-dkms`
  (You should see this line in the output)
  ```
  Removing DKMS module 'ddcci-0.4.1' for kernel-5.xx.yy_z... FAILED!
  ```